components/script/dom/document.rs

```diff
@@ -4125,6 +4125,17 @@ impl Document {
         // Step 6 Run any page visibility change steps which may be defined in other specifications, with visibility
         // state and document. Any other specs' visibility steps will go here.
 
+        // <https://www.w3.org/TR/gamepad/#handling-visibility-change>
+        self.window
```

Comment:
We should only run these if "[visibility state](https://html.spec.whatwg.org/multipage/interaction.html#visibility-state) becomes "hidden""